app-typescript/components/with-pagination.spec.tsx

```diff
@@ -0,0 +1,115 @@
+import {describe, it} from 'mocha';
+import * as assert from 'assert';
+import {mount} from 'enzyme';
+import * as React from 'react';
+import {WithPagination} from './WithPagination';
+import {range} from 'lodash';
+
+interface IPost {
+    title: string;
+}
+
+// Simulate fetching delay
+const TIMEOUT = 1000;
+
+export class Paginated extends React.PureComponent {
+    getItems(): Promise<{items: Array<IPost>, itemCount: number}> {
+        return new Promise((resolve) => {
+            setTimeout(() => {
+                return resolve({items: range(1, 500).map((x) => ({title: `title ${x}`})), itemCount: 100});
+            }, TIMEOUT);
+        });
+    }
+
+    render() {
+        return (
+            <WithPagination
+                getItems={() => this.getItems()}
+            >
+                {
+                    (items) => <div>{JSON.stringify(items)}</div>
+                }
+            </WithPagination>
+        );
+    }
+}
+
+describe.only('with-pagination', () => {
+    it('returns button1 when rendered', (done) => {
```

Comment:
what is button 1 and why should it be rendered? You should do a better job via naming to indicate this. For one, I suggest testing `getPagination` function separately. Then what remains to test in the UI component is to do a simple test for one scenerio to see whether pagination bar renders. Then test behavior - next,prev - specific page and scrolling.